### PR TITLE
Add Trace tuning to doc website repo.

### PR DESCRIPTION
**Pull request by carolxob**
Signed-off-by: carolxob <carolxob@amazon.com>

### Description
Add Trace tuning to doc website repo.

### Issues Addressed
Addresses #1267 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by carolxob** [comment]
@dlvenable This PR is ready for tech review.

**Comment by carolxob** [comment]
@cmanning09 This PR is ready for tech review.

**Comment by carolxob** [comment]
@asifsmohammed This is ready for tech review.

**Comment by carolxob** [comment]
@cwillum This is in technical review, but it also needs doc review.

File: _data-prepper/trace-tuning.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+nav_order: 12
+---
+
+# Trace tuning
+
+Data Prepper for trace analytics in version 0.8.x supports both vertical and horizontal scaling. Adjust the size of your single Data Prepper instance to meet your workload's demands and scale vertically. 
+
+<!--- Is there a better way to phrase the version part of this sentence?--->
+
+Scale horizontally by deploying multiple Data Prepper instances to form a cluster by using the [Core Peer Forwarder](https://github.com/opensearch-project/data-prepper/blob/main/docs/peer_forwarder.md). This enables Data Prepper instances to communicate with instances in the cluster, and is required for horizontally-scaling deployments.
```
Comment (ariamarble) [inline _data-prepper/trace-tuning.md:13]: Maybe say "To scale" or "you can scale" ?

File: _data-prepper/trace-tuning.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+## Scaling tips
+
+The following sections describe useful tips for scaling. We recommend that you modify parameters based on the requirements. We also recommend that you monitor the Data Prepper host metrics and OpenSearch metrics to ensure the configuration is working as expected.
+
+### Buffer
+
+The total number of trace requests that Data Prepper is processing is equal to sum of `buffer_size` in `otel-trace-pipeline` and `raw-trace-pipeline`. The total number of trace requests to OpenSearch is equal to the product of `batch_size` and `workers` in `raw-trace-pipeline`.
```
Comment (ariamarble) [inline _data-prepper/trace-tuning.md:21]: where does `raw-trace-pipeline` exist?

File: _data-prepper/trace-tuning.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+<!--- Are each of these items in code blocks buffers? They look like parameters.--->
```
Comment (ariamarble) [inline _data-prepper/trace-tuning.md:23]: yeah, I agree that they look like parameters for the buffer settings

File: _data-prepper/trace-tuning.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+### Workers 
+
+The `workers` setting determines the number of threads that are used by Data Prepper to process requests from the buffer. We recommend that you set the workers based on the CPU utilization. This value can be higher than available processors, as Data Prepper spends significant input/output time when sending data to OpenSearch.
+
+### Heap
+
+Configure the heap of Data Prepper by setting the `JVM_OPTS` environmental variable. We recommend that you set the heap value to a minimum of `4` * `batch_size` * `otel_send_batch_size` * `maximum size of indvidual span`.
+
+As mentioned in the [setup](trace_setup.md#opentelemetry-collector), set `otel_send_batch_size` to a value of `50` in your opentelemetry collector configuration.
```
Comment (ariamarble) [inline _data-prepper/trace-tuning.md:38]: setup steps? setup instructions?

File: _data-prepper/trace-tuning.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+<!--- This link doesn't work. Is there a current link we can use? --->
+
+### Disk
+
+Data Prepper uses disk to store metadata required for service-map processing, so we recommend only storing the key fields `traceId`, `spanId`, `parentSpanId`, `spanKind`, `spanName` and `serviceName`. The service-map plugin ensures that it only stores two files with each storing `window_duration` seconds of data. During testing, we found that, for a throughput of `3000 spans/second`, the total disk usage was `4 MB`.
```
Comment (ariamarble) [inline _data-prepper/trace-tuning.md:44]: I feel like we should say "local disk" or "local storage" or "disk space" something like that depending on context. Just saying "disk" feels confusing and vague imo

File: _data-prepper/trace-tuning.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+Data Prepper uses the disk to write logs. In the most recent version, you can redirect the logs to the path of your preference.
+
+
+## AWS
+
+The [AWS EC2 Cloudformation](https://github.com/opensearch-project/data-prepper/blob/main/deployment-template/ec2/data-prepper-ec2-deployment-cfn.yaml) template provides a user-friendly mechanism to configure the above scaling attributes.
+
+The [Kubernetes configuration files](https://github.com/opensearch-project/data-prepper/blob/main/examples/dev/k8s/README.md) and [EKS configuration files](https://github.com/opensearch-project/data-prepper/blob/main/deployment-template/eks/README.md) are available to configure these attributes in a cluster deployment.
+
+<!--- Should these links be this long?--->
```
Comment (ariamarble) [inline _data-prepper/trace-tuning.md:55]: the length of the URL shouldn't matter

File: _data-prepper/trace-tuning.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+## Benchmark
+
+We ran tests in a `r5.xlarge` with the following configuration:
```
Comment (ariamarble) [inline _data-prepper/trace-tuning.md:59]:
```suggestion
We ran tests in a `r5.xlarge` EC2 instance with the following configuration:
```

File: _data-prepper/trace-tuning.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+<!--- What is r5.xlarge`? A parameter? A file?--->
```
Comment (ariamarble) [inline _data-prepper/trace-tuning.md:61]: it's an EC2 instance size configuration

**Review by ariamarble — APPROVED** [review]
LGTM other than comments

File: _data-prepper/trace-tuning.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+The `workers` setting determines the number of threads that are used by Data Prepper to process requests from the buffer. We recommend that you set the workers based on the CPU utilization. This value can be higher than available processors, as Data Prepper spends significant input/output time when sending data to OpenSearch.
+
+### Heap
+
+Configure the heap of Data Prepper by setting the `JVM_OPTS` environmental variable. We recommend that you set the heap value to a minimum of `4` * `batch_size` * `otel_send_batch_size` * `maximum size of indvidual span`.
+
+As mentioned in the [setup](trace_setup.md#opentelemetry-collector), set `otel_send_batch_size` to a value of `50` in your opentelemetry collector configuration.
+
+<!--- This link doesn't work. Is there a current link we can use? --->
```
[the remark this answers is not in the document]
Comment (carolxob) [reply _data-prepper/trace-tuning.md:40]: @dlvenable @chenqi0805 Do you have a link for this?

File: _data-prepper/trace-tuning.md [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+
+### Workers 
+
+The `workers` setting determines the number of threads that are used by Data Prepper to process requests from the buffer. We recommend that you set `workers` based on the CPU utilization. This value can be higher than available processors, as Data Prepper spends significant input/output time when sending data to OpenSearch.
+
+### Heap
+
+Configure the heap of Data Prepper by setting the `JVM_OPTS` environmental variable. We recommend that you set the heap value to a minimum of `4` * `batch_size` * `otel_send_batch_size` * `maximum size of indvidual span`.
+
+As mentioned in the [setup setpes](trace_setup.md#opentelemetry-collector), set `otel_send_batch_size` to a value of `50` in your opentelemetry collector configuration.
```
Comment (cwillum) [inline _data-prepper/trace-tuning.md:34]: Is this the right spelling?

"setup setpes"

**Review by cwillum — APPROVED** [review]
Looks good. Just a couple of comments.

File: _data-prepper/trace-tuning.md (hunk 2 above)
Comment (dlvenable) [reply _data-prepper/trace-tuning.md:19]: This is referring to what is now called `raw-pipeline` in https://opensearch.org/docs/latest/data-prepper/pipelines/#trace-analytics-pipeline.

File: _data-prepper/trace-tuning.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (dlvenable) [reply _data-prepper/trace-tuning.md:36]: https://github.com/opensearch-project/data-prepper/blob/main/docs/trace_analytics.md#opentelemetry-collector

[the remark this answers is not in the document]
Comment (dlvenable) [reply _data-prepper/trace-tuning.md:36]: I think the OpenTelemetry Collector page linked above should be included in the Trace Analytics use-case page.

[the remark this answers is not in the document]
Comment (dlvenable) [reply _data-prepper/trace-tuning.md:36]: You are adding this to the documentation in this PR: https://github.com/opensearch-project/documentation-website/pull/1873/files#diff-b82c8183542a303b75efd7945414bc6b977acd2ac7a9f6ecc4b2610430f2d13aR260

File: _data-prepper/trace-tuning.md (hunk 10 above)
Comment (carolxob) [reply _data-prepper/trace-tuning.md:34]: Both of these points are correct.

**Comment by dlvenable** [comment]
This page can be part of #1873, or a sub-page to that page.

**Comment by carolxob** [comment]
Closing this PR because content will be added to #1873.

**Pull request closed without merging**